src/NuGet.Core/NuGet.Commands/SignCommand/CertificateProvider.cs

```diff
@@ -77,7 +77,7 @@ public static async Task<X509Certificate2Collection> GetCertificatesAsync(Certif
                                     options.CertificatePath)));
 
                         case CRYPT_E_NO_MATCH_HRESULT:
-                        case OPENSSL_ERR_R_NESTED_ASN1_ERROR:
+                        case OPENSSL_READER_ASN1_ERROR:
```

Comment:
Since this test is running for both netcoreapp2.1 and netcoreapp5.0, and the HResult only chanegd in netcoreapp5.0, pls differentiate them by #if IS_SIGNING_SUPPORTED && IS_CORECLR

Reply:
From https://github.com/dotnet/runtime/issues/36786#issuecomment-632193883:

> We don't guarantee the HResult values of most of our exceptions, depending on them is going to be fragile.

Even if we fix this, it could break at "any" time.

I noticed that all the errors in the method use NU3001. How important is it that the error messages are different for the different failure scenarios? If we had just a single error message, regardless of `HResult` value, how difficult would it be for customers to know what action to take? Can we just pass through the crypto exception message?  I didn't look at the error messages, so maybe it's obvious. But I'm concerned about using `HResult` at all.

Reply:
I agree that we need to change the code as they don't guarantee the HResult values.
But I think it can be fixed after the insertion. How about raising an issue now so that we won't forget? @kartheekp-ms

Reply:
@zivkan I agree. that is my concern too. Unfortunately there are **7 scenarios** for [NU3001](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu3001). I think we may have to look at all the occurrences of this error code and then modify current implementation. Can we do that in another issue because after modifying code as per @heng-liu [comment](https://github.com/NuGet/NuGet.Client/pull/3405#discussion_r428987004), test case passed on multiple tfms?

Reply:
sure, get things working now, but we'll just be hoping that they don't change between now and when .net 5 ships, and even if the runtime doesn't change before going GA, we'll still be hoping that it doesn't change in any servicing releases. We should have an issue to track improving this.

Reply:
created a tracking issue https://github.com/NuGet/Home/issues/9598